### PR TITLE
DataLinks: Respects display name and adds field quoting

### LINKED ISSUE #26545: DataLinks: Code completion for data links omits field quoting
**What happened**:
Used the data links feature for the first time and used the code completion to understand the syntax for selecting some field:
image

When e.g. selecting `pre.clusterDomain` grafana generates `${__data.fields[pre.clusterDomain]}` into the URL field of the data link.
However I noticed that the replacement doesn't work unless I add quotes around the field name like this `${__data.fields["pre.clusterDomain"]}`

**What you expected to happen**:
Code completion to provide working substitutions.

**Environment**:
- Grafana version: 7.1
- Data source type & version: Elasticsearch 7.x

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
Firstly, this PR switches from using `field.name` to the actual `displayName` for the field. Secondly displayName containing `.` or ` ` are decorated with quotes.

**Which issue(s) this PR fixes**:
Fixes #26545

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/panel/panellinks/link_srv.ts [hunk 1]
```diff
@@ -75,7 +76,7 @@ const valueVars = [
 ];
 
 const buildLabelPath = (label: string) => {
-  return label.indexOf('.') > -1 ? `["${label}"]` : `.${label}`;
+  return label.indexOf('.') > -1 || label.trim().indexOf(' ') > -1 ? `["${label}"]` : `.${label}`;
```
Comment (kaydelaney) [inline public/app/features/panel/panellinks/link_srv.ts:79]: Nit for readability:
```suggestion
  return label.includes('.') || label.trim().includes(' ') ? `["${label}"]` : `.${label}`;
```

**Review by kaydelaney — APPROVED** [review]
One small comment but otherwise LGTM!

**Review by torkelo — APPROVED** [review]
Looks good, but getFrameVars was a bit cryptic, so took some time to parse what was going on :)

**Comment by dprokop** [comment]
Great tests @hugohaggmark!

**Pull request merged**